### PR TITLE
Feature/migration to fabric 1.3

### PR DESCRIPTION
**Pull request by fedyiv**
- replaced fabric docker images with 1.3
- ensured that project build and integration test works

### CODE REVIEW

File: fabric-connector/src/test/java/com/luxoft/fabric/FabricConnectorIntegrationTest.java [hunk 1]
```diff
@@ -24,16 +27,20 @@
...
+        LOG.info("Starting preparation");
         int exitCode = execInDirectory("./fabric.sh restart", "../files/artifacts/");
+        LOG.info("Restarted network");
         Assert.assertEquals(0, exitCode);
 
         fabricConfig = FabricConfig.getConfigFromFile("../files/fabric.yaml");
 
         // Configuring Fabric network
         NetworkManager.configNetwork(fabricConfig);
+        LOG.info("Finished preparation preparation");
```
Comment (osesov) [inline fabric-connector/src/test/java/com/luxoft/fabric/FabricConnectorIntegrationTest.java]: "preparation preparation"?

File: files/chaincode/src/map/map.go [hunk 2]
```diff
@@ -41,27 +42,52 @@ import (
 type SimpleChaincode struct {
 }
 
+func ensureStringIsUTF8Valid(inputString string ) string {
```
Comment (osesov) [inline files/chaincode/src/map/map.go]: it's better not to skip invalid characters, but check if there any invalid and stop processing with error

File: fabric-connector/src/main/java/com/luxoft/fabric/FabricConfig.java [hunk 3]
```diff
@@ -455,20 +508,22 @@ public void installChaincode(HFClient hfClient, List<Peer> peerList, String key)
...
+        logger.info("install chaincode proposal {}:{}", chaincodeID.getName(), chaincodeID.getVersion());
         Collection<ProposalResponse> installProposalResponse = hfClient.sendInstallProposal(installProposalRequest, peerList);
 
         checkProposalResponse("install chaincode", installProposalResponse);
 
         UpgradeProposalRequest upgradeProposalRequest = hfClient.newUpgradeProposalRequest();
         upgradeProposalRequest.setChaincodeID(chaincodeID);
         upgradeProposalRequest.setChaincodeVersion(chaincodeVersion);
-        upgradeProposalRequest.setProposalWaitTime(60000);
+        upgradeProposalRequest.setProposalWaitTime(120000);
```
Comment (osesov) [inline fabric-connector/src/main/java/com/luxoft/fabric/FabricConfig.java:519]: Should we implement option for that?

File: files/chaincode/src/map/map.go (hunk 2 above)
Comment (fedyiv) [reply files/chaincode/src/map/map.go]: This function used in following context:

```
// Init is a no-op
func (t *SimpleChaincode) Init(stub shim.ChaincodeStubInterface) pb.Response {
	creator, err := stub.GetCreator();
	creatorString := ensureStringIsUTF8Valid(string(creator[:]))
```

and it its turn

```
  // GetCreator returns `SignatureHeader.Creator` (e.g. an identity)
  // of the `SignedProposal`. This is the identity of the agent (or user)
  // submitting the transaction.
  GetCreator() ([]byte, error)
```

And this SignedProposal is built by Fabric SDK, so we have no control over this logic and hence it's not possible to fix the test.

I believe this conversion to UTF8 can be considered as normalisation of data on chaincode side

File: fabric-connector/src/main/java/com/luxoft/fabric/FabricConfig.java (hunk 3 above)
Comment (fedyiv) [reply fabric-connector/src/main/java/com/luxoft/fabric/FabricConfig.java:519]: Raised issue in the tracker #9

File: fabric-connector/src/test/java/com/luxoft/fabric/FabricConnectorIntegrationTest.java (hunk 1 above)
Comment (fedyiv) [reply fabric-connector/src/test/java/com/luxoft/fabric/FabricConnectorIntegrationTest.java]: Fixed

File: files/chaincode/src/map/map.go (hunk 2 above)
Comment (osesov) [reply files/chaincode/src/map/map.go]: I still believe it's better to transform byte string to valid string completely, without throwing some bytes away, due to two reasons:
1) there is not only X509Identity in fabric, there are IdeMix identities as well, which is much more complex and have lots of byte strings embedded, so loss might be non-negligiable
2) This behaviour might have changed in the future versions of Fabric (f.e., they can attach binary version of certificate eventually), but we would miss it silently

Thus I propose to use something like url.PathEscape/url.PathUnescape to convert to valid string with not much overhead

Comment (fedyiv) [reply files/chaincode/src/map/map.go]: Fixed

File: files/chaincode/src/map/map.go [hunk 4]
```diff
@@ -44,24 +45,29 @@ type SimpleChaincode struct {
 // Init is a no-op
 func (t *SimpleChaincode) Init(stub shim.ChaincodeStubInterface) pb.Response {
 	creator, err := stub.GetCreator();
-	fmt.Printf("Init creator: %s", string(creator[:]))
+	creatorString := url.PathEscape(string(creator[:]))
```
Comment (osesov) [inline files/chaincode/src/map/map.go]: By the way, I wonder if  for some imaginary reasons, stub.GetCreator returns err, so creator might be bogus or nil, would all of `url.PathEscape(string(creator[:]))` handle that situation correctly?
Probably it worth to move error check earlier?

Comment (fedyiv) [reply files/chaincode/src/map/map.go]: I totally agree. Fixed

**Pull request merged**